reactivesocket-core/src/main/java/io/reactivesocket/server/ReactiveSocketServer.java

```diff
@@ -42,24 +45,28 @@
 
     static ReactiveSocketServer create(TransportServer transportServer) {
         return acceptor -> {
-            return transportServer.start(duplexConnection -> {
-                return Px.from(duplexConnection.receive())
+            return transportServer.start(connection -> {
+                return Px.from(connection.receive())
                   .switchTo(setupFrame -> {
                       if (setupFrame.getType() == FrameType.SETUP) {
+                          ClientServerInputMultiplexer multiplexer = new ClientServerInputMultiplexer(connection);
                           ConnectionSetupPayload setupPayload = ConnectionSetupPayload.create(setupFrame);
-                          ClientReactiveSocket sender = new ClientReactiveSocket(duplexConnection,
+                          ClientReactiveSocket sender = new ClientReactiveSocket(multiplexer.asServerConnection(),
                                                                                  Throwable::printStackTrace,
                                                                                  StreamIdSupplier.serverSupplier(),
                                                                                  KeepAliveProvider.never());
+                          LeaseHonoringSocket lhs = new DefaultLeaseHonoringSocket(sender);
```

Comment:
The oddity here is that there is no way today in the protocol where a server can tell whether it will honor leases from the client. Probably this should piggy back on the client will honor lease in the setup saying if a client honors lease, server will too.
I am interested in hearing what others think about this.